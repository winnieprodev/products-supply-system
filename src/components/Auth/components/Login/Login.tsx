import { Button } from "components";

import * as Styled from "./login.styled";

interface OwnProps {
  onClick: () => void;
}
const Login: React.FC<OwnProps> = ({ onClick }) => (
  <>
    <Styled.Wrapper>
      <Button primary fullWidth onClick={onClick}>
        Login
      </Button>
    </Styled.Wrapper>
    <Styled.Footer />
  </>
);

export default Login;
