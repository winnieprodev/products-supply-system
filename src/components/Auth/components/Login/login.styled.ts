import styled from "styled-components";
import { primaryColor } from "styles/colors";

export const Wrapper = styled.div`
  width: 200px;
  margin: 10vw auto 0;
`;

export const Footer = styled.div`
  position: fixed;
  left: 0;
  right: 0;
  bottom: 0;
  background: ${primaryColor};
  height: 10vh;
`;
