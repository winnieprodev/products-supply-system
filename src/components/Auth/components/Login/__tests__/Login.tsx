import { render } from "@testing-library/react";

import Login from "../Login";

describe("Login", () => {
  describe("Snapshot tests", () => {
    test("it matches the snapshot", () => {
      const { container } = render(<Login onClick={jest.fn()}>test</Login>);
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
