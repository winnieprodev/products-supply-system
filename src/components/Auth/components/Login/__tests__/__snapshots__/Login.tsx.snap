// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Snapshot tests it matches the snapshot 1`] = `
.c1 {
  outline: none;
  cursor: pointer;
  border: 0 none;
  -webkit-transition: background ease 0.3s;
  transition: background ease 0.3s;
  background-color: #F25C05;
  color: #FFFFFF;
}

.c1:hover {
  background-color: #DD5202;
}

.c1:active {
  background-color: #FF7626;
}

.c2 {
  border: 0;
  border-radius: 8px;
  padding: 8px 16px;
  margin: 4px;
  font-weight: 700;
  width: 100%;
  margin: 0;
}

.c0 {
  width: 200px;
  margin: 10vw auto 0;
}

<div
  class="c0"
>
  <button
    class="c1 c2"
  >
    Login
  </button>
</div>
`;
